tianshou/highlevel/optim.py

```diff
@@ -1,20 +1,30 @@
 from abc import ABC, abstractmethod
-from typing import Any, Protocol
+from collections.abc import Iterable
+from typing import Any, Protocol, TypeAlias
 
 import torch
 from torch.optim import Adam, RMSprop
 
 from tianshou.utils.string import ToStringMixin
 
+TParams: TypeAlias = Iterable[torch.Tensor] | Iterable[dict[str, Any]]
+
 
 class OptimizerWithLearningRateProtocol(Protocol):
     def __call__(self, parameters: Any, lr: float, **kwargs: Any) -> torch.optim.Optimizer:
         pass
 
 
 class OptimizerFactory(ABC, ToStringMixin):
+    def create_optimizer(
+        self,
+        module: torch.nn.Module,
+        lr: float,
+    ) -> torch.optim.Optimizer:
+        return self.create_optimizer_for_params(module.parameters(), lr)
+
     @abstractmethod
-    def create_optimizer(self, module: torch.nn.Module, lr: float) -> torch.optim.Optimizer:
+    def create_optimizer_for_params(self, params: TParams, lr: float) -> torch.optim.Optimizer:
```

Comment:
Would it have a downside to have a non-abstract `create_optimizer` here with the previous signature that uses `create_optimizer_for_params`? Then you wouldn't need the downstream changes while still keeping flexibility.

Using the `module.params` is a good default that works for most cases

Reply:
I'm confused, `create_optimizer` still exists, is non-abstract, is implemented via `create_optimizer_for_params` and there are no downstream changes.

Reply:
Sorry, I glanced over the diff too fast and misread it. All good, merging